ksql-parser/src/main/java/io/confluent/ksql/parser/tree/ExpressionTreeRewriter.java

```diff
@@ -381,25 +381,6 @@ public Expression visitFunctionCall(FunctionCall node, Context<C> context) {
       return node;
     }
 
-    @Override
-    protected Expression visitLambdaExpression(LambdaExpression node, Context<C> context) {
```

Comment:
Why have we deleted all this? How does this relate to the PR?

Reply:
We were not supporting Lambda anyways and also since it was using the same characters was confusing the grammar to pick Lambda instead of struct field reference. By removing it we can use the struct field reference correctly and also won't have unused/unsupported feature in the grammar.

Reply:
but isn't this kinda rash ? - why are we precluding adding a cool future feature (not saying it should be in the codebase already, just trying to leave space in the plan for it) simply by being opinionated over our choice of separator char ?
FWIW i already gave feedback that not only does a choice of `->` preclude doing something interesting in the future that we may well want to add, it's also confusing to the target audience: folks from a sql and big data background are used to single-character identifiers, whereas the folks from a java-programming background are going to read it as a lamba.

Reply:
I don't think we're precluding it. It just means that we would have to choose a different syntax for it.

I'm also not _that_ worried about confusing java developers. I would think most java developers have some familiarity with c/c++, so the notion of using a '->' to dereference a struct should feel familiar.

Reply:
Already mentioned to Hojjat that we can use '=>' for lambdas.  '->' is used by a couple of other dbs for referencing struct fields.